README.md
# Yotube
download videos from youtube.com

# Just run pip -r requirements
## run the "run.bat" file and you are all good :)
open the players.json file add something like this :  
"<index start with 0>":" <'number of videos from video page'> ,<https://www.youtube.com/<'your favourite youtuber'>/videos"

example:  
{  
    "0":"5,https://www.youtube.com/c/DailyDoseOfInternet/videos",  
    "1":"1,https://www.youtube.com/c/xQcOW/videos"  
}  
## note:  
if Cookie doesnt work, get the latest cookie from your browser and add it to line 9:  


example:  
### "Cookie": "CONSENT=YES+cb.20210813-05-p0.en+FX+867"


history.txt
https://www.youtube.com/watch?v=XVhaospXEKc
https://www.youtube.com/watch?v=ph2ZBNFzhsQ
https://www.youtube.com/watch?v=XVhaospXEKc
https://www.youtube.com/watch?v=ph2ZBNFzhsQ
https://www.youtube.com/watch?v=1pHDDgorH4M
https://www.youtube.com/watch?v=2j947T7nMxo
https://www.youtube.com/watch?v=U4Z2BgL6AT4
https://www.youtube.com/watch?v=vidn2TWvIYo
https://www.youtube.com/watch?v=XVhaospXEKc
https://www.youtube.com/watch?v=ph2ZBNFzhsQ
https://www.youtube.com/watch?v=1pHDDgorH4M
https://www.youtube.com/watch?v=2j947T7nMxo
https://www.youtube.com/watch?v=U4Z2BgL6AT4
https://www.youtube.com/watch?v=vidn2TWvIYo
https://www.youtube.com/watch?v=XVhaospXEKc
https://www.youtube.com/watch?v=ph2ZBNFzhsQ
https://www.youtube.com/watch?v=1pHDDgorH4M
https://www.youtube.com/watch?v=2j947T7nMxo
https://www.youtube.com/watch?v=U4Z2BgL6AT4
https://www.youtube.com/watch?v=vidn2TWvIYo
https://www.youtube.com/watch?v=jim9rjNvfEA
https://www.youtube.com/watch?v=UQ3KHtuY5-8
https://www.youtube.com/watch?v=pLaV0mOks6I
https://www.youtube.com/watch?v=XVhaospXEKc
https://www.youtube.com/watch?v=ph2ZBNFzhsQ
https://www.youtube.com/watch?v=1pHDDgorH4M
https://www.youtube.com/watch?v=2j947T7nMxo
https://www.youtube.com/watch?v=U4Z2BgL6AT4
https://www.youtube.com/watch?v=vidn2TWvIYo
https://www.youtube.com/watch?v=jim9rjNvfEA
https://www.youtube.com/watch?v=UQ3KHtuY5-8
https://www.youtube.com/watch?v=pLaV0mOks6I
https://www.youtube.com/watch?v=XVhaospXEKc
https://www.youtube.com/watch?v=ph2ZBNFzhsQ
https://www.youtube.com/watch?v=1pHDDgorH4M
https://www.youtube.com/watch?v=2j947T7nMxo
https://www.youtube.com/watch?v=U4Z2BgL6AT4
https://www.youtube.com/watch?v=vidn2TWvIYo
https://www.youtube.com/watch?v=jim9rjNvfEA
https://www.youtube.com/watch?v=UQ3KHtuY5-8
https://www.youtube.com/watch?v=pLaV0mOks6I
https://www.youtube.com/watch?v=XVhaospXEKc
https://www.youtube.com/watch?v=ph2ZBNFzhsQ
https://www.youtube.com/watch?v=1pHDDgorH4M
https://www.youtube.com/watch?v=2j947T7nMxo
https://www.youtube.com/watch?v=U4Z2BgL6AT4
https://www.youtube.com/watch?v=vidn2TWvIYo
https://www.youtube.com/watch?v=jim9rjNvfEA
https://www.youtube.com/watch?v=UQ3KHtuY5-8
https://www.youtube.com/watch?v=pLaV0mOks6I
https://www.youtube.com/watch?v=XVhaospXEKc
https://www.youtube.com/watch?v=ph2ZBNFzhsQ
https://www.youtube.com/watch?v=1pHDDgorH4M
https://www.youtube.com/watch?v=2j947T7nMxo
https://www.youtube.com/watch?v=U4Z2BgL6AT4
https://www.youtube.com/watch?v=vidn2TWvIYo
https://www.youtube.com/watch?v=jim9rjNvfEA
https://www.youtube.com/watch?v=UQ3KHtuY5-8
https://www.youtube.com/watch?v=pLaV0mOks6I
https://www.youtube.com/watch?v=XVhaospXEKc
https://www.youtube.com/watch?v=ph2ZBNFzhsQ
https://www.youtube.com/watch?v=1pHDDgorH4M
https://www.youtube.com/watch?v=2j947T7nMxo
https://www.youtube.com/watch?v=U4Z2BgL6AT4
https://www.youtube.com/watch?v=vidn2TWvIYo
https://www.youtube.com/watch?v=jim9rjNvfEA
https://www.youtube.com/watch?v=UQ3KHtuY5-8
https://www.youtube.com/watch?v=pLaV0mOks6I
https://www.youtube.com/watch?v=XVhaospXEKc
https://www.youtube.com/watch?v=ph2ZBNFzhsQ
https://www.youtube.com/watch?v=1pHDDgorH4M
https://www.youtube.com/watch?v=2j947T7nMxo
https://www.youtube.com/watch?v=U4Z2BgL6AT4
https://www.youtube.com/watch?v=vidn2TWvIYo
https://www.youtube.com/watch?v=jim9rjNvfEA
https://www.youtube.com/watch?v=UQ3KHtuY5-8
https://www.youtube.com/watch?v=pLaV0mOks6I
https://www.youtube.com/watch?v=V99Hp8VNbcM
https://www.youtube.com/watch?v=XVhaospXEKc
https://www.youtube.com/watch?v=ph2ZBNFzhsQ
https://www.youtube.com/watch?v=1pHDDgorH4M
https://www.youtube.com/watch?v=2j947T7nMxo
https://www.youtube.com/watch?v=V99Hp8VNbcM
https://www.youtube.com/watch?v=vidn2TWvIYo
https://www.youtube.com/watch?v=jim9rjNvfEA
https://www.youtube.com/watch?v=UQ3KHtuY5-8
https://www.youtube.com/watch?v=pLaV0mOks6I
https://www.youtube.com/watch?v=vXufjSNypK8
https://www.youtube.com/watch?v=jg4zHEMWzbw
https://www.youtube.com/watch?v=sPThdzBBWC8
https://www.youtube.com/watch?v=1IkXwOOgN8U
https://www.youtube.com/watch?v=J5m1V-05ZDg
https://www.youtube.com/watch?v=4sXWhYF5gME
https://www.youtube.com/watch?v=wOPGcA7ua00
https://www.youtube.com/watch?v=VABmqGzUNWo
https://www.youtube.com/watch?v=1-KAAaUdBxU
https://www.youtube.com/watch?v=-zUJzB35yaY
https://www.youtube.com/watch?v=232495m_Sag
https://www.youtube.com/watch?v=AuBxCuik5Pg
https://www.youtube.com/watch?v=OPwWA-ZtA5w
https://www.youtube.com/watch?v=W88cX31fXV4
https://www.youtube.com/watch?v=EeTg2rhb_pM
https://www.youtube.com/watch?v=iMw4UwkqAEw


YotubeD.py
import requests as req
import json
import re as Regex

regex = r"/watch\?v=[\w\d_-]+"
Headers = {
    "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.102 Safari/537.36",
    "accept": "text/html,application/json",
    "Cookie": "CONSENT=YES+cb.20210813-05-p0.en+FX+867"
}
print("+loading config ...")
with open('players.json', encoding="utf8") as json_data:
    players = json.load(json_data)
print("+Clearing output ...")
with open('output.bat', mode="w", encoding="utf8") as file:
    file.write("")

baseURL = "https://www.youtube.com"

print("+loading history ...")
with open('history.txt', mode="r", encoding="utf8") as file:
    history = file.readlines()

print("+opening a session...")
session = req.session()
session.headers = Headers

for i in players:
    datalink = str(players[i]).split(",")
    count = int(datalink[0])
    data = session.get(datalink[1])
    print("getting links :"+str(data.status_code))
    links = Regex.findall(regex, data.text)
    for i in range(0, count):
        link = baseURL + links[i]
        enable = True
        for L in history:
            M = L.replace("\n", "")
            if link == M:
                enable = False
                break
        if enable:
            print("hit a video")
            with open('output.bat', mode="a", encoding="utf8") as file:
                file.write("pytube "+link+"\n")
            with open('history.txt', mode="a", encoding="utf8") as file:
                file.write(link+"\n")
session.close()
